lib/bolt/project_migrator/modules.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+require 'bolt/project_migrator/base'
+
+module Bolt
+  class ProjectMigrator
+    class Modules < Base
+      def migrate(project, modulepath)
+        return true unless ENV['BOLT_MODULE_FEATURE'] && project.modules.nil?
+
+        @outputter.print_message "Migrating project modules\n\n"
+
+        config     = project.project_file
+        puppetfile = project.puppetfile
+        moduledir  = project.managed_moduledir
+
+        # Notify user to manually migrate modules if using non-default modulepath
+        if modulepath != project.modulepath
+          @outputter.print_migrate_step(
+            "Project has a non-default configured modulepath, unable to automatically "\
+            "migrate project modules. To migrate project modules manually, see "\
+            "http://pup.pt/bolt-modules"
+          )
+          true
+        # Migrate modules from Puppetfile
+        elsif File.exist?(puppetfile)
+          migrate_modules_from_puppetfile(config, puppetfile, moduledir, modulepath)
```

Comment:
It seems safer to pass `project.modulepath` here :sparkles: just in case :sparkles:

Reply:
This is still 'modulepath' it looks like?

Reply:
In thinking more about this, I can't think of a scenario where we want to use this for any modulepath other than `['modules', 'site-modules', 'site']`. Setting that as a constant in this class and using it directly instead of passing the modulepath around would be more clear, would simplify the signatures, and would prevent us from accidentally operating on any other modulepaths unintentionally. I think it'd also make the class more readable and easier to follow.

Reply:
Obvious caveat we still need to check that the user modulepath is that modulepath, and exit if not. I think `moduledir` can also be hardcoded as 'modules' ?

Reply:
I think using a hardcoded modulepath is fine here, but I don't think having it as a constant simplifies much, as the directories need to be expanded relative to the project path.

Updated.